### PR TITLE
[changelog-generator] Fix original commit resolving

### PR DESCRIPTION
**Pull request by alloy**
Fixes https://github.com/react-native-community/releases/pull/166/files#r387227509 & https://github.com/react-native-community/releases/pull/166/files#r387227231

### CODE REVIEW

File: scripts/changelog-generator.ts [hunk 1]
```diff
@@ -246,17 +246,16 @@ export function git(gitDir: string, ...args: string[]) {
...
   const match = item.commit.message.match(/Differential Revision: (D\d+)/m);
   if (match) {
-    const drev = match[1];
     return git(
       gitDir,
       "log",
       "master",
       "--pretty=format:%H",
-      `--grep=${drev}`
+      `--grep=${match[0]}`
```
Comment (alloy) [inline scripts/changelog-generator.ts]: This was too ambiguous, as some commit messages refer to these differential revisions too.

File: scripts/changelog-generator.ts [hunk 2]
```diff
@@ -246,17 +246,16 @@ export function git(gitDir: string, ...args: string[]) {
...
     ).then(sha => {
```
Comment (TheSavior) [inline scripts/changelog-generator.ts]: Do you want to add an assert that there is only one match?

File: package.json [hunk 3]
```diff
@@ -5,7 +5,8 @@
   "scripts": {
     "format": "prettier *.md scripts{,/**}/*.js --write",
     "format:check": "prettier *.md scripts{,/**}/*.js --list-different",
-    "generate": "ts-node --transpile-only ./scripts/changelog-generator.ts",
```
Comment (TheSavior) [inline package.json]: What does this change do?

Comment (alloy) [reply package.json]: This makes it possible to have scripts that specify `node` options, such as the `generate:debug` command that allows attaching a debugger. When `$_NODE_OPTS` is undefined the command is technically identical to the previous version.

**Comment by alloy** [comment]
Alright, shipping it :shipit:

**Pull request merged**